src/containers/Post/PostDetail.jsx

```diff
@@ -3,23 +3,45 @@ import { Segment } from 'semantic-ui-react';
 import TabBar from '../../components/TabBar/TabBar';
 import StartMenu from '../../components/StartMenu/StartMenu';
 import PostDetailComponent from '../../components/Post/PostDetailComponent';
+import NoMatch from '../../components/NoMatch/NoMatch';
 
 class PostDetail extends Component {
   state = {
     tabBarName: localStorage.getItem('username'),
+    loading: false
+  }
+  
+  checkPost = async () => {
+    const response = await fetch(`${process.env.REACT_APP_DEV_API_DOMAIN}/posts/${this.props.match.params.post_id}`);
+    if (response.status !== 404) {
+      this.setState({ loading: true });
+    }
+    else if (response.status === 404) {
+      console.log('page not found');
+      this.setState({ loading: false });
+    }
```

Comment:
I would get rid of the `console.log(...)` and don't use `else if(...)` if there's really only 2 conditions.